src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -38,7 +38,7 @@ internal abstract class JsonPropertyInfo
         public string NameUsedToCompareAsString { get; private set; }
 
         // The escaped name passed to the writer.
-        public byte[] EscapedName { get; private set; }
+        public JsonEncodedText? EscapedName { get; private set; }
```

Comment:
nit: Why nullable? Instead of `.HasValue`, we could just check if `EscapedName.Equals(default)` (to check if it isn't set).